walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,35 +27,42 @@
   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
 
-    ArrayList<TimeRange> allUnavailableTimes = new ArrayList<>();
+    ArrayList<TimeRange> mandatoryUnavailableTimes = new ArrayList<>();
+    ArrayList<TimeRange> optionalUnavailableTimes = new ArrayList<>();
 
-    for (Event event : events){
-      if (!isEventBlockingRequest(event, request)){
-        allUnavailableTimes.add(event.getWhen());
+    for (Event event : events) {
+
+      if (!isEventBlockingRequest(event, request, true)) {
+        mandatoryUnavailableTimes.add(event.getWhen());
       }
-    }
 
-    ArrayList<TimeRange> mergedUnavailableTimes = mergeTimeRanges(allUnavailableTimes);
-    ArrayList<TimeRange> availableTimes = new ArrayList<>();
-    
-    TimeRange possibleTime;
-    int prevEndTime = TimeRange.START_OF_DAY;
-    
-    for (TimeRange curTime: mergedUnavailableTimes){
+      if(!isEventBlockingRequest(event, request, false)) {
+        optionalUnavailableTimes.add(event.getWhen());
+      }
+    }
 
-      possibleTime = TimeRange.fromStartEnd(prevEndTime, curTime.start(), false);
+    ArrayList<TimeRange> allUnavailableTimes;
+    boolean combinedOptionalMandatoryTimes = false;
```

Comment:
combinedOptionalMandatoryTimes is kind of redundant and a mouth full. Perhaps choose either combinedTimes or OptionalAndMandatoryTimes